src/functionModules/components/PurchaseProcessModule.ts

```diff
@@ -74,12 +74,18 @@ export default class PurchaseProcessModule {
       store,
     )
 
-    const [owner, bytesRequested] = [delivery[0], delivery[1]]
+    const owner = delivery[0]
+
+    const isDelivered = await DatatrustContractModule.isDelivered(
+      ethereum.selectedAddress,
+      deliveryHash,
+      ethereum.selectedAddress,
+      store,
+    )
 
     // owner is coming back with some letters uppercase for some reason?
     const isBuyer = ethereum.selectedAddress === owner.toLowerCase()
```

Comment:
this needs to be compared using `web3.utils.toChecksumAddress(ethereum.selectedAddress)`

Reply:
to be addressed later